File: JavaScript/Math-Array-String/80.js
// 1512. Number of Good Pairs


var numIdenticalPairs = function(nums) {
    let countMap = {};  // To store frequency of each number
    let goodPairs = 0;

    for (let num of nums) {
        if (countMap[num]) {
            goodPairs += countMap[num]; // Add existing count to goodPairs
            countMap[num]++; // Increase the count
        } else {
            countMap[num] = 1; // Initialize count
        }
    }

    return goodPairs;
};

// Example test cases
console.log(numIdenticalPairs([1,2,3,1,1,3])); // Output: 4
console.log(numIdenticalPairs([1,1,1,1]));     // Output: 6
console.log(numIdenticalPairs([1,2,3]));       // Output: 0

File: Python/Math-Array-String/80.py
# 1512. Number of Good Pairs

from typing import Counter, List

class Solution:
    def numIdenticalPairs(self, nums: List[int]) -> int:
        count = Counter(nums)  # Count occurrences of each number
        good_pairs = sum((val * (val - 1)) // 2 for val in count.values())  # Compute pairs
        return good_pairs
    

sol = Solution()
output = sol.numIdenticalPairs([1,1,1,1])
print(output)  # Output: 6